src/platform/mock-local-home-platform.ts

```diff
@@ -1,144 +1,91 @@
-  private mockNetwork: MockNetwork | undefined;
   private deviceManager: smarthome.DeviceManager = new DeviceManagerStub();
   private app: AppStub | undefined;
   private localDeviceIds: Map<string, string> = new Map<string, string>();
   private newDeviceRegisteredActions: ((localDeviceId: string) => void)[] = [];
-  private homeAppReady: boolean = false;
 
   private constructor() {}
 
-  public initializeRadio(
-    mockNetwork: MockNetwork,
-    udpScanConfigs: UDPScanConfig[]
-  ) {
-    this.mockNetwork = mockNetwork;
-    this.udpScanConfigs = udpScanConfigs;
-    this.setupUDP();
-  }
-
   public setApp(app: AppStub) {
     this.app = app;
   }
 
-  private onNewDeviceIdRegistered(localDeviceId: string) {
-    this.newDeviceRegisteredActions.forEach(newDeviceRegisteredAction => {
-      newDeviceRegisteredAction(localDeviceId);
-    });
-  }
-
-  public async getNextDeviceIdRegistered(): Promise<string> {
-    return new Promise(resolve => {
-      this.newDeviceRegisteredActions.push(localDeviceId => {
-        resolve(localDeviceId);
-      });
-    });
-  }
-
-  public isHomeAppReady(): boolean {
-    return this.homeAppReady;
+  public getDeviceManager(): smarthome.DeviceManager {
+    return this.deviceManager;
   }
 
-  public notifyHomeAppReady(): void {
-    this.homeAppReady = true;
+  public getLocalDeviceIdMap(): Map<string, string> {
+    return this.localDeviceIds;
   }
 
-  //  Singleton getter
-  public static getInstance(): MockLocalHomePlatform {
-    if (!MockLocalHomePlatform.instance) {
+  /**
+   * Returns the static singleton instance, creating it if needed.
+   * @param resetState  Whether or not to force a new instance.  Useful for initializing tests.
+   */
+  public static getInstance(
+    resetState: boolean = false
+  ): MockLocalHomePlatform {
+    if (!MockLocalHomePlatform.instance || resetState) {
       MockLocalHomePlatform.instance = new MockLocalHomePlatform();
     }
     return MockLocalHomePlatform.instance;
   }
 
-  public getDeviceManager(): smarthome.DeviceManager {
-    return this.deviceManager;
-  }
-
-  public getLocalDeviceIdMap(): Map<string, string> {
-    return this.localDeviceIds;
+  private onNewDeviceIdRegistered(localDeviceId: string) {
+    this.newDeviceRegisteredActions.forEach(newDeviceRegisteredAction => {
+      newDeviceRegisteredAction(localDeviceId);
+    });
   }
 
-  private setupUDP() {
-    this.udpScanConfigs.forEach(udpScanConfig => {
-      this.mockNetwork!.registerUDPListener(
-        this,
-        udpScanConfig.listenPort,
-        udpScanConfig.broadcastAddress
-      );
+  /**
+   * Asyncronously returns the next localDeviceId registered to the Local Home Platform.
+   * This localDeviceId is referred to as the verificationId in the IdentifyResponse
+   */
+  public async getNextDeviceIdRegistered(): Promise<string> {
+    return new Promise(resolve => {
+      this.newDeviceRegisteredActions.push(localDeviceId => {
```

Comment:
can't simply push resolve?

Reply:
Could you show me what you mean?  Currently using this constructor so the action has access to `resolve`

Reply:
I was able to remove this function and the action entirely by just returning the local device id in the `triggerIdentify` call